ui/app/components/variable-form.js

```diff
@@ -326,4 +329,53 @@ export default class VariableFormComponent extends Component {
       this.args.model.pathLinkedEntities?.task
     );
   }
+
+  //#region Unsaved Changes Confirmation
+
+  @computed(
+    'args.model.{keyValues,path}',
+    'keyValues.@each.{key,value}',
+    'path'
+  )
+  get hasUserModifiedAttributes() {
+    const compactedBasicKVs = this.keyValues
+      .map((kv) => ({ key: kv.key, value: kv.value }))
+      .filter((kv) => kv.key || kv.value);
+    const compactedPassedKVs = this.args.model.keyValues.filter(
+      (kv) => kv.key || kv.value
+    );
+    const unequal =
+      !isEqual(compactedBasicKVs, compactedPassedKVs) ||
+      !isEqual(this.path, this.args.model.path);
+    return unequal;
+  }
+
+  addExitHandler() {
+    this.router.on('routeWillChange', this, this.confirmExit);
+  }
+
+  removeExitHandler() {
+    this.router.off('routeWillChange', this, this.confirmExit);
+  }
```

Comment:
`comment`: If its good for [Ballint](https://balinterdi.com/blog/warning-about-losing-unsaved-changes-in-ember-octane/), then its good for us.